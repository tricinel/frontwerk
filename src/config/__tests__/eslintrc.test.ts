/* eslint-disable global-require */

import jestSerializerPath from 'jest-serializer-path';

import {
  unquoteSerializer,
  winPathSerializer
} from '../../helpers/serializers';

expect.addSnapshotSerializer(unquoteSerializer);
expect.addSnapshotSerializer(winPathSerializer);
expect.addSnapshotSerializer(jestSerializerPath);

afterEach(() => {
  jest.resetModules();
});

test('Gets the eslint config when there is Jest as a project dependency', () => {
  jest.mock('../../utils/hasDep', () => ({
    hasDep: (dep: string): boolean => dep === 'jest'
  }));
  const config = require('../eslintrc');
  expect(config.env.jest).toBe(true);
  expect(config.plugins).toContain('jest');
  expect(config.extends).toContain('plugin:jest/recommended');
  expect(config.extends).toEqual(
    expect.arrayContaining([expect.stringContaining('eslint-config-frontwerk')])
  );

  expect(config).toMatchSnapshot();
});

test('Gets the eslint config when there is React as a project dependency', () => {
  jest.mock('../../utils/hasDep', () => ({
    hasDep: (dep: string): boolean => dep === 'react'
  }));
  const config = require('../eslintrc');
  expect(config.env.jquery).toBe(false);
  expect(config.extends).toEqual(
    expect.arrayContaining([
      expect.stringContaining('eslint-config-frontwerk-react')
    ])
  );

  expect(config).toMatchSnapshot();
});

test('Gets the eslint config when there is TypeScript as a project dependency', () => {
  jest.mock('../../utils/hasDep', () => ({
    hasDep: (dep: string): boolean => dep === 'typescript'
  }));
  const config = require('../eslintrc');
  expect(config.env.jquery).toBe(false);
  expect(config.extends).toEqual(
    expect.arrayContaining([
      expect.stringContaining('eslint-config-frontwerk-typescript'),
      expect.stringContaining('prettier/@typescript-eslint')
    ])
  );

  expect(config).toMatchSnapshot();
});

test('Gets the eslint config when there are Jest and jQuery as project dependencies', () => {
  jest.mock('../../utils/hasDep', () => ({
    hasDep: (dep: string): boolean => dep === 'jest' || dep === 'jquery'
  }));
  const config = require('../eslintrc');
  expect(config.env.jest).toBe(true);
  expect(config.env.jquery).toBe(true);
  expect(config.plugins).toContain('jest');

  expect(config).toMatchSnapshot();
});
