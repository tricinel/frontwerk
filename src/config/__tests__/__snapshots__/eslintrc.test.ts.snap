// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gets the eslint config when there are Jest and jQuery as project dependencies 1`] = `
Object {
  env: Object {
    browser: true,
    jest: true,
    jquery: true,
    node: true,
  },
  extends: Array [
    <PROJECT_ROOT>/node_modules/eslint-config-frontwerk/index.js,
    <PROJECT_ROOT>/node_modules/eslint-config-prettier/index.js,
    plugin:prettier/recommended,
    plugin:jest/recommended,
  ],
  parserOptions: Object {},
  plugins: Array [
    jest,
  ],
  rules: Object {
    import/no-extraneous-dependencies: Array [
      error,
      Object {
        devDependencies: true,
      },
    ],
  },
}
`;

exports[`Gets the eslint config when there is Jest as a project dependency 1`] = `
Object {
  env: Object {
    browser: true,
    jest: true,
    jquery: false,
    node: true,
  },
  extends: Array [
    <PROJECT_ROOT>/node_modules/eslint-config-frontwerk/index.js,
    <PROJECT_ROOT>/node_modules/eslint-config-prettier/index.js,
    plugin:prettier/recommended,
    plugin:jest/recommended,
  ],
  parserOptions: Object {},
  plugins: Array [
    jest,
  ],
  rules: Object {
    import/no-extraneous-dependencies: Array [
      error,
      Object {
        devDependencies: true,
      },
    ],
  },
}
`;

exports[`Gets the eslint config when there is React as a project dependency 1`] = `
Object {
  env: Object {
    browser: true,
    jest: true,
    jquery: false,
    node: true,
  },
  extends: Array [
    <PROJECT_ROOT>/node_modules/eslint-config-frontwerk-react/index.js,
    <PROJECT_ROOT>/node_modules/eslint-config-prettier/index.js,
    plugin:prettier/recommended,
    plugin:jest/recommended,
  ],
  parserOptions: Object {},
  plugins: Array [
    jest,
  ],
  rules: Object {
    import/no-extraneous-dependencies: Array [
      error,
      Object {
        devDependencies: true,
      },
    ],
    react/jsx-filename-extension: Array [
      error,
      Object {
        extensions: Array [
          .jsx,
        ],
      },
    ],
  },
}
`;

exports[`Gets the eslint config when there is TypeScript as a project dependency 1`] = `
Object {
  env: Object {
    browser: true,
    jest: true,
    jquery: false,
    node: true,
  },
  extends: Array [
    <PROJECT_ROOT>/node_modules/eslint-config-frontwerk-typescript/index.js,
    prettier/@typescript-eslint,
    <PROJECT_ROOT>/node_modules/eslint-config-prettier/index.js,
    plugin:prettier/recommended,
    plugin:jest/recommended,
  ],
  parserOptions: Object {
    project: ./tsconfig.json,
    tsconfigRootDir: <PROJECT_ROOT>,
  },
  plugins: Array [
    jest,
  ],
  rules: Object {
    import/no-extraneous-dependencies: Array [
      error,
      Object {
        devDependencies: true,
      },
    ],
  },
}
`;
